### ISSUE TITLE
BracesAroundHashParameters flagging necessary braces as redundant

### ISSUE THREAD

**Issue by rlue** [rubocop-hq/rubocop#5212]
In a Rails project, have a hash parameter with a proc as one of its values:

```ruby
accepts_nested_attributes_for :brands, {
  allow_destroy: true,
  reject_if: proc do |attributes|
    attributes.values_at(:name, :photo, :id).all?(&:blank?)
  end
}
```

Because of the block used to define the proc, the braces around the hash parameter are necessary. Without them, instantiating this model results in the following error:

```
>> foo = Account.first
ArgumentError: tried to create Proc object without a block
```

However, the `BracesAroundHashParameters` cop is flagging the braces as redundant:

```
app/models/account.rb:49:42: C: Redundant curly braces around a hash parameter.
  accepts_nested_attributes_for :brands, { ...
```

Is there another approach / recommended style for this scenario, or is this an issue with the cop?

---

Rubocop v0.51.0, Rails v5.1.4, Ruby 2.4.1p111

**Comment by rlue** [comment]
Apparently all warnings go away if I use curly braces for the block:

```ruby
accepts_nested_attributes_for :brands, allow_destroy: true,
                                       reject_if: proc { |attributes|
                                         attributes
                                           .values_at(:name, :photo, :id)
                                           .all?(&:blank?)
                                       }
```

Rubocop does not raise any warnings for curly braces on a multi-line block here. Is this an accepted/recommended stylistic approach to this case?

**Comment by chuck-john** [comment]
I can confirm this behavior. The following throws a `Style/BracesAroundHashParameters` violation:

```ruby
... do |app|
  Capybara::Selenium::Driver.new app, {
    browser:              :chrome,
    http_client:          Selenium::WebDriver::Remote::Http::Default.new(read_timeout: 90),
    desired_capabilities: Selenium::WebDriver::Remote::Capabilities.chrome(
	  ...
    )
  }
end
```

environment
---
* Rubocop 0.54.0
* Ruby 2.4.4

**Comment by rlue** [comment]
Bump

**Comment by rlue** [comment]
Bump
